app/services/EventSinkService.scala

```diff
@@ -41,20 +29,18 @@ class EventSinkService {
   val queueName = Play.configuration.getString(EventSinkService.QUEUE_NAME_CONFIG_KEY)
     .getOrElse(EventSinkService.QUEUE_NAME_DEFAULT_VALUE)
 
-  def logEvent(category: String, metadata: JsValue) = {
-    Logger.info("eventsink.exchangename=" + exchangeName)
-    Logger.info("eventsink.queueName=" + queueName)
-
-    Logger.info("Submitting message to event sink exchange: " + Json.stringify(metadata))
-
-    //val message = EventSinkMessage(Instant.now().getEpochSecond, category, metadata)
-    messageService.submit(exchangeName, queueName, metadata, "fanout")
+  def logEvent(message: JsValue) = {
+    // Inject timestamp before logging the event
+    val event = message.as[JsObject] + ("created" -> Json.toJson(java.util.Date.from(Instant.now())))
+    Logger.info("Submitting message to event sink exchange: " + Json.stringify(event))
+    messageService.submit(exchangeName, queueName, event, "fanout")
```

Comment:
```suggestion
    try {
      messageService.submit(exchangeName, queueName, event, "fanout")
    } catch {
      case e: Throwable => { Logger.error("Failed to submit event sink message", e) }
    }
```

Giving this one final look, I'm noticing that the user can't log in if RabbitMQ is not running. The exception doesn't cause Clowder to crash, but prevents a log in simply because the event cannot be logged.

If RabbitMQ is still an optional dependency, we might consider catching these exceptions with something like the above.

Reply:
if MessageService is not wired to Rabbitmq it will call the `noop` function which will log something. I think the pattern is to add a call to `connect()` in every Rabbitmq call that makes sure it is connected to the broker (see other function bodies in the Rabbitmq implementation). Can we add that to this `submit` implementation?

No matter what, I am not sure what will happen if `connect()` fails. It might still fail and this catch would still be needed.